spec/airport_spec.rb

```
@@ -0,0 +1,134 @@
+  # expect(new_airport.storm_landing(new_flight)).to eq("Lufthansa did not land.")
+  # end
+
+  # it 'check if plane is included in Airport' do
+  # new_airport.stub(:storm_decision) {"no"}
+  # expect(new_airport.storm_landing(new_flight)).to eq("Plane is already in the airport, cannot land")
+  # end
+
+  # it 'check that if user input for storm decision is yes the plane is landed' do
+  #  allow(new_airport).to receive(:storm_decision).and_return("yes")
+  #  expect(new_airport.planes_in_airport).to eq([new_flight.name])
+  # end
+
+  it 'airport reponds to plane_already_in_airport?' do
+    expect(new_airport).to respond_to(:plane_already_in_airport)
+  end
+
+  it 'airport responds to storm_landing with one argument' do
+    expect(new_airport).to respond_to(:plane_already_in_airport).with(1).argument
+  end
+
+  it 'check if plane included in airport' do
+    expect(new_airport.plane_already_in_airport(new_flight)).to eq("No")
+  end
+
+  it 'check if plane landed is added to array' do
+    new_airport.land_plane(new_flight)
+    expect(new_airport.planes_in_airport).to eq([new_flight.name])
+  end
+
+  it 'check if plane in airport outputs "yes" when planes_already_in_airport method called' do
+    new_airport.land_plane(new_flight)
+    expect(new_airport.plane_already_in_airport(new_flight)).to eq("Yes")
+  end
+
+  it 'airport responds to take_off method' do
+    expect(new_airport).to respond_to(:take_off)
+  end
+
+  it 'airport responds to storm_landing with one argument' do
+    expect(new_airport).to respond_to(:take_off).with(1).argument
+  end
+
+  it 'check if plane removed from airport' do
+    new_airport.land_plane(new_flight)
+    new_airport.take_off(new_flight)
+    expect(new_airport.planes_in_airport).to eq([])
+  end
+
+  it 'if attempt is make to take off plane not in airport receive error' do
+    expect(new_airport.take_off(new_flight)).to eq("Plane cannot take off as it is not in the airport")
```

Ruby and rspec have a different way of handling errors. Here I would have used the rspec syntax:

expect { code }.to raise_error "Cannot take off".

Which would have required you to use the command "raise" in the Ruby code, rather than returning a string.

https://rollbar.com/guides/ruby-raising-exceptions/